src/modules/upload/UploadQueue.jsx

```diff
@@ -14,6 +14,17 @@ import {
 import getMimeTypeIcon from '@/lib/getMimeTypeIcon'
 
 export const DumbUploadQueue = translate()(props => {
+  const { successCount, purgeQueue, queue, doneCount } = props
+
+  useEffect(() => {
+    if (successCount == doneCount && successCount === queue?.length) {
+      const timer = setTimeout(() => {
+        purgeQueue()
+      }, 5000)
```

Comment:
LGTM, just a small change:

we can store this value in a constant in `src/constants/config.js`

```suggestion
      }, DEFAULT_UPLOAD_PROGRESS_HIDE_DELAY)
```